src/auth/auth-config.ts

```diff
@@ -15,102 +15,138 @@
+ */
 export interface SAMLAuthProviderConfig extends AuthProviderConfig {
+  /**
+   * The SAML IdP entity identifier.
+   */
   idpEntityId: string;
+  /**
+   * The SAML IdP SSO URL. This must be a valid URL.
+   */
   ssoURL: string;
+  /**
+   * The list of SAML IdP X.509 certificates issued by CA for this provider.
+   * Multiple certificates are accepted to prevent outages during
+   * IdP key rotation (for example ADFS rotates every 10 days). When the Auth
+   * server receives a SAML response, it will match the SAML response with the
+   * certificate on record. Otherwise the response is rejected.
+   * Developers are expected to manage the certificate updates as keys are
+   * rotated.
+   */
   x509Certificates: string[];
+  /**
+   * The SAML relying party (service provider) entity ID.
+   * This is defined by the developer but needs to be provided to the SAML IdP.
+   */
   rpEntityId: string;
+  /**
+   * This is fixed and must always be the same as the OAuth redirect URL
+   * provisioned by Firebase Auth,
+   * `https://project-id.firebaseapp.com/__/auth/handler` unless a custom
+   * `authDomain` is used.
+   * The callback URL should also be provided to the SAML IdP during
+   * configuration.
+   */
   callbackURL?: string;
+  /**
+   * @internal
+   */
   enableRequestSigning?: boolean;
 }
 
-/** The server side SAML configuration request interface. */
-export interface SAMLConfigServerRequest {
-  idpConfig?: {
-    idpEntityId?: string;
-    ssoUrl?: string;
-    idpCertificates?: Array<{
-      x509Certificate: string;
-    }>;
-    signRequest?: boolean;
-  };
-  spConfig?: {
-    spEntityId?: string;
-    callbackUri?: string;
-  };
-  displayName?: string;
-  enabled?: boolean;
-  [key: string]: any;
-}
-
-/** The server side SAML configuration response interface. */
-export interface SAMLConfigServerResponse {
-  // Used when getting config.
-  // projects/${projectId}/inboundSamlConfigs/${providerId}
-  name?: string;
-  idpConfig?: {
-    idpEntityId?: string;
-    ssoUrl?: string;
-    idpCertificates?: Array<{
-      x509Certificate: string;
-    }>;
-    signRequest?: boolean;
-  };
-  spConfig?: {
-    spEntityId?: string;
-    callbackUri?: string;
-  };
-  displayName?: string;
-  enabled?: boolean;
-}
-
-/** The server side OIDC configuration request interface. */
-export interface OIDCConfigServerRequest {
-  clientId?: string;
-  issuer?: string;
-  displayName?: string;
-  enabled?: boolean;
-  [key: string]: any;
-}
-
-/** The server side OIDC configuration response interface. */
-export interface OIDCConfigServerResponse {
-  // Used when getting config.
-  // projects/${projectId}/oauthIdpConfigs/${providerId}
-  name?: string;
-  clientId?: string;
-  issuer?: string;
-  displayName?: string;
-  enabled?: boolean;
-}
-
 /** The public API response interface for listing provider configs. */
 export interface ListProviderConfigResults {
```

Comment:
Isn't there any documentation for this type in the existing typings?

Reply:
Thanks, missed it!